### PR TITLE
render: Implement linear RGB interpolation for gradients (fix #598)

### LINKED ISSUE #598: Linear RGB setting is not implemented for gradients
Visual test: https://github.com/ruffle-rs/visual-tests/tree/master/tests/simple_shapes/gradients/focal_radial

Expected:
image

Actual:
image

Reproduced with wgpu on desktop, and webgl on web.

### PR DESCRIPTION
**Pull request by Herschel**
This implements the "Linear RGB" interpolation setting for gradients on all backends.

 * For linear space gradients, colors are converted from sRGB space to linear space before being stored as uniforms/SVG data.
 * On 3d backends, the shader is tweaked to convert back to sRGB after interpolating.
 * On `wgpu`, the order of the `GradientUniforms` struct was adjusted to avoid issues with alignment.
 * For canvas, there is an SVG attribute `color-interpolation`, but it seems to have no effect/no browsers support it (https://bugzilla.mozilla.org/show_bug.cgi?id=298281). Instead, a `feComponentTransfer` filter is used to apply a gamma curve to the path.

### CODE REVIEW

File: render/wgpu/shaders/gradient.frag [hunk 1]
```diff
@@ -6,18 +6,27 @@ layout(set = 0, binding = 2) uniform Colors {
 };
 
 layout(std430, set = 0, binding = 3) buffer Gradient {
+    vec4 u_colors[16];
+    float u_ratios[16];
     int u_gradient_type;
     uint u_num_colors;
     int u_repeat_mode;
+    int u_interpolation;
```
Comment (Dinnerbone) [inline render/wgpu/shaders/gradient.frag:14]: I feel like the smart thing to do would be to merge `u_gradient_type` and `u_repeat_mode` and `u_interpolation` into a bitmask and save us a few bytes & alignment issues. I don't know if it's worth doing right now though, this works!

Later, `u_colors` and `u_ratios` will either move into a texture or into a variable length `Record u_records[]`, which will need to be at the end of the array. But later :D

Comment (Justin-CB) [reply render/wgpu/shaders/gradient.frag:14]: Wouldn't the code to deal with a bitmask add more bytes than using a bitmask would save? I don't know too much about rust, so I could be completely wrong.

Comment (Dinnerbone) [reply render/wgpu/shaders/gradient.frag:14]: I mean bytes in the buffer, which is more about memory transfer cost. In my experience, keep frequently transferred buffers tiny - but that's video games and I don't know how much that's a real world issue here yet.

It does mean more code, so bytes on filesize sure. That's not too much of an issue here. But for bytes in memory:
Let's say there's 100 gradients in a movie (a reasonable number I think?).
It costs 4 bytes per int, so it's 24 bytes per gradient to have individual type, repeat mode, and interpolation. For 100 gradients that's 2,400 bytes of memory. With only 1 int to contain all three, that's 400 bytes of memory.

It's trivial stuff and maybe it's too early to optimise such things, but it's an easy win to keep in mind!

Comment (Justin-CB) [reply render/wgpu/shaders/gradient.frag:14]: Oh, OK, that makes sense. I didn't realize that the buffer the flags were on would be duplicated so many times.

**Comment by Herschel** [comment]
I agree, it'd be nice to do an optimization pass over the shaders in the near future. (The annoying one here will be dealing with WebGL1, as usual, with no easy bitwise operators).

**Pull request merged**